test/mpi/coll/test_coll_algos.sh

```diff
@@ -262,4 +262,30 @@ for algo_name in ${algo_names}; do
     done
 done
 
+######### Add tests for Alltoallv algorithms ###########
+
+#disable device collectives for alltoallv to test MPIR algorithms
+testing_env="env=MPIR_CVAR_ALLTOALLV_DEVICE_COLLECTIVE=0 "
+
+#test nb algorithms
+testing_env+="env=MPIR_CVAR_ALLTOALLV_INTRA_ALGORITHM=nb "
+testing_env+="env=MPIR_CVAR_IALLTOALLV_DEVICE_COLLECTIVE=0 "
+algo_names="gentran_scattered"
+batchsizes="1 2 4"
+outstandingtasks="4 8"
+for algo_name in ${algo_names}; do
+    env="${testing_env} env=MPIR_CVAR_IALLTOALLV_INTRA_ALGORITHM=${algo_name} "
+    if [ ${algo_name} -eq "gentran_scattered"]; then
+        for task in $ {outstandingtasks}; do
+            for batchsize in $ {batchsizes}; do
+                env="${testing_env} env=MPIR_CVAR_IALLTOALLV_INTRA_ALGORITHM=${algo_name} "
+                env+="env=MPIR_CVAR_IALLTOALLV_SCATTERED_BATCH_SIZE=${batchsize} "
+                env+="env=MPIR_CVAR_IALLTOALLV_SCATTERED_OUTSTANDING_TASKS=${task} "
+                 coll_algo_tests+="alltoallv 8 ${env}${nl}"
+                 coll_algo_tests+="alltoallv0 10 ${env}${nl}"
+            done
+        done
+    fi
+done
+
```

Comment:
This tests `MPI_Alltoall`, not `MPI_Ialltoall`, right?

Reply:
No it does not test `MPI_Ialltoall`. This is a non blocking schedule based implementation for `MPI_Alltoallv` collective. We set CVARs to follow this code path.

Reply:
@tarudoodi I am a bit confused that whether you are agreeing with me or saying something different, LOL.

If the test code: `test/mpi/coll/alltoallv.c` only tests `MPI_Alltoallv` and this PR is for `MPI_Ialltoallv`, then we'll need add a `ialltoallv.c` to the test.

Reply:
This is not a PR for `MPI_Ialltoallv` its for a `MPI_Alltoallv`. the PR was correctly titled before you renamed it. This code is called from `MPI_Alltoallv` with the help of following CVARs. We force it use the `non blocking` path.
`export MPIR_CVAR_ALLTOALL_DEVICE_COLLECTIVE=0`
`export MPIR_CVAR_IALLTOALL_DEVICE_COLLECTIVE=0`
`export MPIR_CVAR_ALLTOALL_INTRA_ALGORITHM=nb`
`export MPIR_CVAR_IALLTOALL_INTRA_ALGORITHM=gentran_scattered`